demos/graph-classification/supervised-graph-classification.ipynb

```diff
@@ -0,0 +1,559 @@
+{
```

Comment:
<p>The validation accuracy is quite quantised here, which might be surprising/disconcerting to some readers (I certainly was a bit wary of it, until I worked out what was going on). Is it worth mentioning the quantisation is because the validation/test set for each fold has only ~19 (10% of 188) elements?</p><p> </p><p>Also, it seems that training barely does anything for the metrics on the train set, especially given how much noise there is?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1053/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Reply:
I think the accuracy plot is not surprising given, as you said, the small amount of data. Also, the plot is for the best found model out of 50 checked models so we don't know exactly how the losses for all the other models behave although I suspect similarly. The dataset is also not balanced; there are twice as many graphs labeled "1" than "-1".

Generally, the loss is the more interesting quantity here although this being the best model, it looks like it got really lucky because the validation loss is lower than the training loss. It could be that this batch is easy to predict. The training loss look like it behaves correctly although it is noisy. We can make it look smoother by reducing the learning rate. The value I currently use was a good compromise between learning fast enough and producing reasonably looking loss curves.

Anyway, these curves don't always look like the ideal one we see in textbooks. If we could run this on a bigger dataset with some more interesting and informative features, and well balanced, I suspect the curves would look more textbook-like.

I'll mention a couple of the above points in the notebook.

Reply:
Looks great.